### PR TITLE
fix: improve ipynb loading hitches

### PR DESCRIPTION
**Pull request by ashtonG**
this fixes three issues that happen when loading an ipynb file in the code editor:

1) when an ipynb file was selected and the file content wasn't loaded, the error handler was called.

2) when a parse error occurred, the error handler was called with incorrect arguments.

3) when loading an ipynb file, the unparsed file content renders to the page briefly, even when the content is valid.

We also update the loadable usage in the codeeditor to use the loadable class methods and prevent the filename info from shrinking and growing during the file loading process.

### CODE REVIEW

**Review by EmilyBonar — APPROVED** [review]
Looks good, though personally I find the `pipe`s harder to read.

**Comment by ashtonG** [comment]
good point -- i think it's useful to represent the result as an `Either` (as opposed to `string | error`), but the data-last functions and piping can be hard to follow. i'll add comments explaining what's going on for posterity.
